devenv/lib/repository.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import annotations
+
+import os.path
+from configparser import ConfigParser
+
+from devenv.lib.config import get_config
+from devenv.lib.fs import ensure_binroot
+
+
+class Repository:
+    path: str
+    name: str
+    config_path: str
+
+    def __init__(self, root: str) -> None:
+        self.path = root
+        self.name = root.split("/")[-1]
+
+        # .devenv may exist but not be config_path because .devenv/bin is hard-coded, so we check for config.ini
+        if os.path.exists(f"{root}/.devenv/config.ini"):
+            self.config_path = f"{root}/.devenv"
+        # devenv _may_ exist but not be config_path because devenv has a devenv module, so we check for config.ini
+        elif os.path.exists(f"{root}/devenv/config.ini"):
+            self.config_path = f"{root}/devenv"
+        else:
+            # new default config_path is .devenv
+            self.config_path = f"{root}/.devenv"
+
+    def __repr__(self) -> str:
+        return self.config_path
```

Comment:
`__repr__` should contain a programmer-specific representation of an object, for example `Repository('whatever/path')` -- if this class extends `typing.NamedTuple` you'll get all this for free